actions/profile/edit.php

```diff
@@ -42,23 +42,27 @@ function profile_array_decoder(&$v) {
 		$value = _elgg_html_decode($value);
 	}
 
+	// convert tags fields to array values
+	if ($valuetype == 'tags') {
+		$value = string_to_tag_array($value);
+	}
+	
 	// limit to reasonable sizes
-	// @todo - throwing away changes due to this is dumb!
-	// ^^ This is a sticky form so changes aren't lost...?
-	if (!is_array($value) && $valuetype != 'longtext' && elgg_strlen($value) > 250) {
-		$error = elgg_echo('profile:field_too_long', array(elgg_echo("profile:{$shortname}")));
-		register_error($error);
-		forward(REFERER);
+	if ($valuetype != 'longtext') {
+		$check_values = (array) $value;
+		// also check tags/checkboxes/etc
+		foreach ($check_values as $v) {
+			if (elgg_strlen($v) > 250) {
```

Comment:
This is probably another ticket, but I always found this limit of 250 arbitrary and the profile seems to be the only place we impose it. Perhaps we should make it a global constant and impose the limit once we have better validation logic in place